### PR TITLE
ci: encore PR titles to follow the conventional-commit rules

### LINKED ISSUE #2569: [Feature Request]: A GitHub action to validate the title of PR
### What problem does this feature solve?

Since we are using commit messages of `main` branch for releasing and generating changelog, we need an action to validate the title of PR to enforce a good formated PR title.

### What does the proposed API of configuration look like?

.

### PR DESCRIPTION
**Pull request by hyf0**
## Summary

```
feat(ui): Add `Button` component
^    ^    ^
|    |    |__ Subject
|    |_______ Scope
|____________ Type
```

This PR validate

- Must write a `Type`
  - allowing types are https://github.com/web-infra-dev/rspack/compare/hyf_lint_pr?expand=1#diff-fc98139577eb07d54e24aa5d41da856c636f5ad318d0ad700f13a457c136cb80R23-R32
- Optionally write a `Scope`
  - `Scope` should be lower-cases
- Must write a `Subject`

## Related issue (if exists)

Closes #2569.

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
I think it might be better to add this requirement to `CONTRIBUTING.md`

**Comment by hyf0** [comment]
I need your guys' approvals due to codeowners @hardfist @h-a-n-a cc

**Review by hardfist — APPROVED** [review]

**Pull request merged**
